gateway/src/apicast/gc.lua

```diff
@@ -0,0 +1,53 @@
+-- In LuaJIT and Lua 5.1, the __gc metamethod does not work in tables, it only
+-- works in "userdata". This module introduces a workaround to make it work
+-- with tables.
+
+local getmetatable = getmetatable
+local newproxy = newproxy
+local ipairs = ipairs
+local pairs = pairs
+local tostring = tostring
+
+local _M = {}
+
+local function __gc(proxy)
+  local t = getmetatable(proxy).__table
+  local mt = getmetatable(t)
+
+  if mt and mt.__gc then mt.__gc(t) end
+end
+
+local function __call(proxy, ...)
+  local t = getmetatable(proxy).__table
```

Comment:
I think we should defined this as:

```lua
getmetatable(proxy).__table(...)
```

Because now it basically defines all tables callable.

Unfortunately that can yield rather cryptic error message: `attempt to call field '__table' (a nil value)`.

Unfortunately Lua tries to be helpful and the error message depends on how you call the table:
* `attempt to call local 'varname' (a table value)`
* `attempt to call a table value`

So replicating those messages might not be possible.

Based on how the function is stored (or if it is a field).
The only workaround I can imagine is using `pcall` like:

```lua
local function __call(self, ...)
  local mt = rawgetmetatable(self)
  local ret = { pcall(mt.__table, ...) }

  local ok = table.remove(ret, 1)

  if ok then
	  return unpack(ret)
  else
	  error(ret[1], 2)
  end
end
```

That would throw error: `ERROR: gc.lua:66: attempt to call a table value` (even with correct line pointing to the original caller).